### PR TITLE
Move Default Domain Backend to Postgres

### PR DESCRIPTION
**Pull request by gcapalbo**
Adds a migration for moving the default SMS backend setting for a domain to the Postgres model for backend mappings. Probably easiest to review all at once rather than each commit separately.

@NoahCarnahan @snopoke code review buddies

### CODE REVIEW

File: corehq/apps/sms/models.py [hunk 1]
```diff
@@ -1676,3 +1677,7 @@ def has_migration_completed(cls, name):
             return True
         except cls.DoesNotExist:
             return False
+
+
+# Import signal receiver so that it gets registered
+from corehq.apps.sms.signals import sync_default_backend_mapping
```
Comment (dimagimon) [inline corehq/apps/sms/models.py]: F401 'sync_default_backend_mapping' imported but unused

File: corehq/apps/sms/management/commands/migrate_domain_default_backend.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        if not balance_only:
+            try:
+                _sync_default_backend_mapping(domain)
+            except Exception as e:
+                print ("Error while syncing domain %s, backend %s: %s" %
+                       (domain.name, domain.default_sms_backend_id, e))
+
+    balance(couch_count)
+    if not balance_only:
+        MigrationStatus.set_migration_completed(MigrationStatus.MIGRATION_DOMAIN_DEFAULT_BACKEND)
```
Comment (snopoke) [inline corehq/apps/sms/management/commands/migrate_domain_default_backend.py]: Is this meant to happen if if the balances don't match?

File: corehq/apps/domain/models.py [hunk 3]
```diff
@@ -880,6 +880,22 @@ def delete(self):
...
+        # Ok to keep these as bulk deletes since all domain-related couch docs
+        # are deleted via bulk delete in self._pre_delete() (so we don't need the
+        # couch-sql sync to run on each object we're deleting)
+        SMS.objects.filter(domain=self.name).delete()
+        MessagingSubEvent.objects.filter(parent__domain=self.name).delete()
+        MessagingEvent.objects.filter(domain=self.name).delete()
+        SelfRegistrationInvitation.objects.filter(domain=self.name).delete()
+        SQLMobileBackendMapping.objects.filter(is_global=False, domain=self.name).delete()
+        MobileBackendInvitation.objects.filter(domain=self.name).delete()
+        SQLMobileBackend.objects.filter(is_global=False, domain=self.name).delete()
```
Comment (snopoke) [inline corehq/apps/domain/models.py]: These should rather be added as delete operations: https://github.com/dimagi/commcare-hq/blob/31feaadf7232f506f7b76723cbaef2f12e27b422/corehq/apps/domain/deletion.py#L55-L55

File: corehq/apps/sms/management/commands/migrate_domain_default_backend.py (hunk 2 above)
Comment (gcapalbo) [inline corehq/apps/sms/management/commands/migrate_domain_default_backend.py]: yes, it's up to the person running the migration (me) to resolve any inconsistencies before continuing, or accept them if there's nothing to do about it

File: corehq/apps/domain/models.py (hunk 3 above)
Comment (gcapalbo) [inline corehq/apps/domain/models.py]: is that for performance or organization? it seems like code-wise the `ModelDeletion`s are doing the same thing

Comment (gcapalbo) [inline corehq/apps/domain/models.py]: (will update it in any case, just curious)

File: corehq/apps/sms/models.py [hunk 4]
```diff
@@ -1676,3 +1678,7 @@ def has_migration_completed(cls, name):
...
```
Comment (dimagimon) [inline corehq/apps/sms/models.py]: F401 'sync_default_backend_mapping' imported but unused

**Comment by gcapalbo** [comment]
@snopoke updated

File: corehq/apps/domain/models.py (hunk 3 above)
Comment (snopoke) [inline corehq/apps/domain/models.py]: It's just for organisation and also allows selectively running them if the
app isn't installed eg. During locally running tests

Comment (gcapalbo) [inline corehq/apps/domain/models.py]: cool, makes sense. build failure looks unrelated, kicked it

File: corehq/apps/sms/models.py (hunk 4 above)
Comment (dimagimon) [inline corehq/apps/sms/models.py]: F401 'sync_default_backend_mapping' imported but unused

**Pull request merged**